cluster.go

```diff
@@ -51,7 +51,7 @@ type member struct {
 }
 
 // newCluster returns a new marathon cluster
-func newCluster(client *http.Client, marathonURL string) (*cluster, error) {
+func newCluster(client *http.Client, marathonURL string, DCOSPath string) (*cluster, error) {
```

Comment:
The location you referenced calls `apiGet`, which in turn calls `apiCall`. All other `api*` should behave similarly.

So everything should go through `apiCall` and my previous suggestion should hopefully work. Or did I miss your point?